src/sentry/processing/backpressure/health.py

```diff
@@ -0,0 +1,61 @@
+from typing import Mapping
+
+import sentry_sdk
+from django.conf import settings
+
+from sentry import options
+from sentry.processing.backpressure.topology import CONSUMERS
+from sentry.utils import redis
+
+
+def _prefix_key(key_name: str) -> str:
+    return f"bp1:{key_name}"
+
+
+UNHEALTHY_KEY_NAME = "unhealthy-consumers"
+
+
+def _unhealthy_consumer_key(name: str) -> str:
+    return _prefix_key(f"{UNHEALTHY_KEY_NAME}:{name}")
+
+
+service_monitoring_cluster = redis.redis_clusters.get(
+    settings.SENTRY_SERVICE_MONITORING_REDIS_CLUSTER
+)
+
+
+def is_consumer_healthy(consumer_name: str = "default") -> bool:
+    """Checks whether the given consumer is healthy by looking it up in Redis.
+
+    NB: If the consumer is not found in Redis, it is assumed to be healthy.
+    This behavior might change in the future.
+    """
+
+    if not options.get("backpressure.checking_enabled"):
+        return True
+    # check if queue is healthy by pinging Redis
+    try:
+        # We set the key if the queue is unhealthy. If the key exists,
+        # the queue is unhealthy and we need to return False.
+        healthy = not service_monitoring_cluster.exists(_unhealthy_consumer_key(consumer_name))
```

Comment:
Can be moved to a follow-up or offline: If for any reason our monitor doesn't run, we should probably consider the pipeline unhealthy. I'd rather invert the logic therefore and look for the existence of a _positive_ key.